openlibrary/plugins/upstream/addbook.py

```diff
@@ -699,9 +699,22 @@ def read_subject(subjects):
 
         # ignore empty authors
         work.authors = [a for a in work.get('authors', []) if a.get('author', {}).get('key', '').strip()]
+        work.authors = self.deduplicate_authors(work.get('authors', []))
 
         return trim_doc(work)
 
+    @staticmethod
+    def deduplicate_authors(authors):
```

Comment:
Huge ♥ for making this maintain original order! ~75% of the time I see a dedupe function in python, it loses the original order :P 

We do have a helper for this though; use the `uniq` function from `helpers.py`